src/Target.cpp

```diff
@@ -307,6 +308,31 @@ bool lookup_arch(const std::string &tok, Target::Arch &result) {
     return false;
 }
 
+const std::map<std::string, Target::Processor> processor_name_map = {
+    {"tune_generic", Target::Processor::ProcessorGeneric},
+    {"tune_k8", Target::Processor::K8},
+    {"tune_k8_sse3", Target::Processor::K8_SSE3},
+    {"tune_amdfam10", Target::Processor::AMDFam10},
+    {"tune_btver1", Target::Processor::BtVer1},
+    {"tune_bdver1", Target::Processor::BdVer1},
+    {"tune_bdver2", Target::Processor::BdVer2},
+    {"tune_bdver3", Target::Processor::BdVer3},
+    {"tune_bdver4", Target::Processor::BdVer4},
+    {"tune_btver2", Target::Processor::BtVer2},
+    {"tune_znver1", Target::Processor::ZnVer1},
+    {"tune_znver2", Target::Processor::ZnVer2},
+    {"tune_znver3", Target::Processor::ZnVer3},
```

Comment:
Important design consideration: currently, the string key is effectively identical to the LLVM CPU string,
and i would really really like to keep it that way, so the proper `tune_*` can be autogenerated easily
from the LLVM CPU string.

Reply:
Want to go ahead and put that in a comment here or on the enum?

Reply:
SGTM on both counts